cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -568,30 +634,30 @@ static PyObject *iterate(PyObject *self, PyObject *args)
 static PyObject *next(PyObject *self, PyObject *args)
 {
     COCOTB_UNUSED(self);
-    gpi_iterator_hdl hdl;
+    gpi_hdl_Object<gpi_iterator_hdl>* hdl_obj;
     gpi_sim_hdl result;
     PyObject *res;
 
-    if (!PyArg_ParseTuple(args, "O&", gpi_iterator_hdl_converter, &hdl)) {
+    if (!PyArg_ParseTuple(args, "O!", python_type_for<decltype(hdl_obj)>(), &hdl_obj)) {
         return NULL;
     }
 
-    // It's valid for iterate to return a NULL handle, to make the Python
+    // TODO:eric-wieser: It's valid for iterate to return a NULL handle, to make the Python
     // intuitive we simply raise StopIteration on the first iteration
```

Comment:
Do we with the code commented out below? Please file a follow-up issue and reference it here if needed, it's easier to track than TODO comments.

Reply:
Will fix, I've worked out the root issue, and it's just documented badly.